.github/workflows/build.yaml

```diff
@@ -0,0 +1,64 @@
+name: Test Build and publish
+
+env:
+  BRANCH: ${{ github.ref_name }}
+  VERSION: 3.4.0
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build-spark:
+    name: Build project
+    runs-on: ubuntu-22.04
+    timeout-minutes: 90
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 8
+      - name: maven-settings-xml-action
+        uses: whelk-io/maven-settings-xml-action@v21
+        with:
+          repositories: '[{ "id": "jfrog", "url": "${{ secrets.REPO_URL }}" }]'
+          plugin_repositories: '[{ "id": "jfrogplugin", "url": "${{ secrets.REPO_URL }}" }]'
+          servers: '[{ "id": "jfrog", "username": "${{ secrets.USERNAME }}" , "password": "${{ secrets.PASSWORD }}" }]'
+          mirrors: '[{ "id": "jfrog", "mirrorOf": "*", "url": "${{ secrets.REPO_URL }}", "name": "jfrog" }]'
+
+      - name: Checkout spark from upstream
+        run: git clone https://github.com/apache/spark.git; cd spark; git checkout tags/v3.4.0
+      - name: Build from source
+        run: cd spark; ./dev/make-distribution.sh --name custom-spark --pip --tgz -Phive -Phive-thriftserver -Pmesos -Pyarn -Pkubernetes
```

Comment:
what if we would compile for kubernetes only? I don't think we are planning to use this on mesos or yarn. Just also to make sure (with the SNAP and ROCK tests) that this indeed works with a more minimal installation